### PR TITLE
Feat: Add e2e tests

### PR DESCRIPTION
**Pull request by adolfoportilla**

### CODE REVIEW

File: authentication_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	mockAuthAPI(exchangeURL, s.auth.clientID, s.auth.clientSecret, mockResponse)
+
+	res, err := s.auth.ExchangeRefreshToken(context.TODO(), &ExchangeRefreshTokenParams{})
+
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), expectedResponse.Access, res.Access)
+	assert.Equal(s.T(), expectedResponse.Refresh, res.Refresh)
+	assert.Equal(s.T(), expectedResponse.ExpiresIn, res.ExpiresIn)
+	assert.NotEmpty(s.T(), res.AccessExpiry)
```
Comment (sankethkatta) [inline authentication_e2e_test.go]: should we be verifying anything additional here? like that it is 2 hrs or 60 days past the current time? or do you think that would essentially rewrite the functionality in the test?

File: vehicle_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+	mockVehicleAPI(mockURL, s.vehicle.accessToken, s.responseHeaders, mockResponse)
+
+	res, err := s.vehicle.GetVIN(context.TODO())
+
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), expectedResponse, res)
+}
+
+func (s *VehicleE2ETestSuite) TestLockE2E() {
+	mockStatus := "1234DLFAJ4"
```
Comment (sankethkatta) [inline vehicle_e2e_test.go]: Modify to be more realistic on what the status is.
```suggestion
	mockStatus := "success"
```

File: vehicle_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+	mockVehicleAPI(mockURL, s.vehicle.accessToken, s.responseHeaders, mockResponse)
+
+	res, err := s.vehicle.Lock(context.TODO())
+
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), expectedResponse, res)
+}
+
+func (s *VehicleE2ETestSuite) TestUnlockE2E() {
+	mockStatus := "1234DLFAJ4"
```
Comment (sankethkatta) [inline vehicle_e2e_test.go]: Modify to be more realistic on what the status is.
```suggestion
	mockStatus := "success"
```

File: authentication_e2e_test.go (hunk 1 above)
Comment (adolfoportilla) [reply authentication_e2e_test.go]: I decided not to test in here that since I already do that in the unit tests!

**Pull request merged**